client/app/services/group/currentGroup.service.js

```diff
@@ -14,7 +14,18 @@ export default class CurrentGroup {
     Object.assign(this, {
       value: {},
       User,
-      SessionUser
+      SessionUser,
+      map: {
```

Comment:
Based on the design principle that controller actions should only be triggered by angular's digest cycle, it makes sense to use watches.

Re observer: Obviously this is a special case with only one observer allowed. So in this regard, the watches would be better since they allow any number :)

I agree that if it would only be about the deregistering, it wouldn't be a huge deal. It's more about having similar interfaces (if possible) to reduce complexity. I would find a data store that is being watched easier to understand than the proxy methods.